src/Nox.Types/Types/TranslatedText/TranslatedText.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System.Globalization;
+using System;
+using System.Runtime.InteropServices;
+
+namespace Nox.Types;
+
+/// <summary>
+/// The translated text class.
+/// </summary>
+public sealed class TranslatedText : ValueObject<(CultureCode Locale, string Text), TranslatedText>
+{
+    public TranslatedTextTypeOptions _translatedTextTypeOptions = new();
+
+    /// <summary>
+    /// Gets the text.
+    /// </summary>
+    public string Text
+    {
+        get => Value.Text;
+        private set => Value = (Value.Locale, value);
+    }
+
+    /// <summary>
+    /// Gets the locale.
+    /// </summary>
+    public CultureCode Locale
+    {
+        get => Value.Locale;
+        private set => Value = (value, Value.Text);
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TranslatedText"/> class.
+    /// </summary>
+    public TranslatedText() { Value = (CultureCode.From("en-US"), string.Empty); }
+
+
+    /// <summary>
+    /// Creates an instance of  <see cref="TranslatedText"/> class From the provided parameters.
+    /// </summary>
+    /// <param name="locale">The locale.</param>
+    /// <param name="translatedText">The translated text.</param>
+    /// <param name="translatedTextTypeOptions">The translated text type options.</param>
+    /// <returns>A TranslatedText.</returns>
+    public static TranslatedText From(CultureCode locale, string translatedText, TranslatedTextTypeOptions translatedTextTypeOptions)
```

Comment:
The default From method accepts a tuple (Locale, Text) as parameter. 
Maybe we should standardize the constructors and change this one to accept a tuple and options?